### ISSUE TITLE
Add support for Technogym Run

### ISSUE THREAD

**Issue by MrCsabaToth** [TrackMyIndoorWorkout/TrackMyIndoorWorkout#461]
This is not the MyRun, but the newest treadmill called simply "Run".
I came across one at SDC23. The trick is that it advertises itself as an FTMS Treadmill, however it never provides standard `Treadmill Data`. It does have the RSC companion sensor service exposed though and that transmits. So we'd need to cherry pick and distinguish this particular treadmill, and then use only that sensor. If the app tries to handle it as a treamdill it stays mute, nothing happens. Due to this behavior standard apps like GymTrakr also mute got tricked and remain mute - I guess waiting for the "main" data as me.

I've been able to test some code (although some Samsung folks were very fearful when they realized I was debugging, next time I shouldn't be such a smart ass and vocal, just do the debug as secretively as possible).

I must add that using the Treadmill with Samsung Health is suuuper easy: just touch the Samsung watch at the NFC location and then it automatically connects and everything is recorded. The FTMS way is not overly hard, but not as convenient for sure.

**Comment by MrCsabaToth** [comment]
![IMG_20231005_143657](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/84f662e1-f492-4126-b37e-6c635db1cfe8)
![IMG_20231005_143637](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/903c032e-193b-483c-b565-bd438cc9a44f)

**Comment by MrCsabaToth** [comment]
Some logs: [DebugLogs20231006T003451129937Z.txt](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/files/12837003/DebugLogs20231006T003451129937Z.txt)

**Comment by MrCsabaToth** [comment]
nRF Connect screenshots:
![Screenshot_2023-10-05-18-01-03-47_3c2e989be58abe9ca7463ffb1e6967d2](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/9d1ba01a-5d73-4443-a50f-20307564c188)

![Screenshot_2023-10-05-18-01-14-56_3c2e989be58abe9ca7463ffb1e6967d2](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/33161271-e9ed-4713-80a8-7d6701504aa7)

![Screenshot_2023-10-05-18-01-54-88_3c2e989be58abe9ca7463ffb1e6967d2](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/9d897dc5-062e-4081-804a-e5566b39ebed)

![Screenshot_2023-10-05-18-02-17-64_3c2e989be58abe9ca7463ffb1e6967d2](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/ebeb56cc-51e9-4ba1-af16-fd21b4cb9a46)

![Screenshot_2023-10-05-18-02-39-38_3c2e989be58abe9ca7463ffb1e6967d2](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/0f31bf94-adb0-43a3-b1b5-8fe7b11433be)

![Screenshot_2023-10-05-18-03-47-98_3c2e989be58abe9ca7463ffb1e6967d2](https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/assets/744949/5f7e4f6d-5810-406f-9c5b-658c326c0888)

**Comment by MrCsabaToth** [comment]
I realized I must check model to not collide with Technogym's MyRun trainer, because that's the same manufacturer, but its Treadmill Data service is not mute.

**Comment by MrCsabaToth** [comment]
One last piece would be to add a preferences setting with which this workaround could be blocked or enforced. Name could be 'rsc_only_ftms_treadmill'.

**Comment by MrCsabaToth** [comment]
Got released with the latest beta (build number 150)

**Comment by MrCsabaToth** [comment]
After the `Google Cloud Live: Bay Area` (https://inthecloud.withgoogle.com/google-cloud-live-bay-area/register.html at the MTV campus's new Bay View 300 building) I stopped by the SDC24 (Samsung Developer Conference 24). This was possible because in 2024 it was not in the Moscone but rather the San Jose Convention Center (same as where the Intel Innovate was on 2023). Lot of ChargePoint there in the parking lot (at Google you cannot charge as a visitor, or it is extremely cumbersome) and participated in code labs.

Some were IoT / Smart Home. Had to install a driver on a smart hub and could control a Hue bulb. Another setup used Gen AI to phrase to create smart scenes: like if I press that smart button turn that bulb red, or if the window open sensor triggers then turn the bulb on. The Gen AI version deducted these (trigger event + target smart device + action on the device) from human text or voice. I spoke with the engineer of that system and they had two LLMs in that agentic system:
1. Large LLM to interpret and preprocess the human text
2. Small specialized LLM fine tuned to output JSON to write these which translates to the scene scripts.
3. They used sanitized (anonimized) data to generate synthetic training data.

Another set of labs were about Samsung Health. One set up a health study server in a Docker container setup, then deployed a mobile app and a wearable app. The Samsung Galaxy watch was measuring health metrics (heart rate, SpO2, ECG), before that the study was established, the user was enrolled via the mobile app. Then the wear app synced to the mobile app, which synced to the study server, where we could display graphs.

The lab engineers were from South Korea, and this engineer and his boss saw that my pulse was 60. I told them I'll run Tokyo marathon in 2025 and they were excited. Here are some photos:
![IMG_20241003_171244977](https://github.com/user-attachments/assets/a0cd49c1-3b15-4bad-bc0f-551e05e89b6c)

ECG data:
![IMG_20241003_171441276](https://github.com/user-attachments/assets/2212c618-288d-4240-90ad-668e1f2c2e89)

![IMG_20241003_171512635](https://github.com/user-attachments/assets/0ebf6a61-0e76-493c-b1c5-63b7fe624144)

**Comment by MrCsabaToth** [comment]
So about SDC24 https://news.samsung.com/global/sdc24, I'm telling this is because after the labs closed and there was the party and raffles I went into the cordoned area to verify the Technogym Run support (and completed some company training videos while walking). Fortunately it works:
https://www.strava.com/activities/12577086872

![IMG_20241003_174525253_HDR](https://github.com/user-attachments/assets/70738e11-ce54-44f2-8d6b-e885fa02a62d)

![IMG_20241003_174612784_HDR](https://github.com/user-attachments/assets/400daf5f-573d-43f3-827e-fa8c51eeb743)

**Comment by cagnulein** [comment]
so they finally put ftms on their device by default?

**Comment by MrCsabaToth** [comment]
> so they finally put ftms on their device by default?

Well, this was possible a year ago as well, when I came across the Technogym Run first time (at SDC23), maybe it was exactly this one. Note that the MyRun is different than the Run. Since the FTMS Treadmill Data doesn't have cadence metric, it is usually treated by also exposing an RSC (Running Cadence Sensor) profile. What is special with the Run is that it not only exposes an RSC profile, it doesn't really communicate via the FTMS Treadmill Data. However I can get what I want from the RSC. So a year ago that needed a change because I never considered the RSC as a "primary" or "main" profile / sensor, and I needed to prepare the logic to that scenario.

Maybe I should have double checked that it still doesn't send data through the FTMS Treadmill Data, because you never know when things change due to a firmware update. Over the years as I'm trying to tackle Life Fitness first their own app even wasn't able to handle their own machines, now that works.
